### PR TITLE
Enhancement/Every process needs a log

### PR DESCRIPTION
**Pull request by tkah**
* **Problem:**
   1. Main and render processes log their information to the same file, which can cause some confusion/inconvenience when debugging.
* **Solution:**
   1. Separate the `logger` into `mainLogger` and `renderLogger`, each pointing to different file.
* **Tests:**
   1. `npm run test` in base directory
   1. `npm run test:e2e` in coinstac-ui package directory.
   1. Check that two separate log files exist in the log directory.

### CODE REVIEW

File: packages/coinstac-client-core/src/index.js [hunk 1]
```diff
@@ -89,7 +89,7 @@ class CoinstacClient {
     this.appDirectory = opts.appDirectory ||
       CoinstacClient.getDefaultAppDirectory();
     this.halfpennyBaseUrl = opts.hp;
-    this.logger = opts.logger || new Logger({ transports: [new Console()] });
+    this.logger = opts.mainLogger || new Logger({ transports: [new Console()] });
```
Comment (swashcap) [inline packages/coinstac-client-core/src/index.js:92]: We should keep this as `logger` so it’s more general. _coinstac-client-core_ should be environment-agnostic – unaware that it's running in a “main” Electron process.

File: packages/coinstac-ui/app/render/state/ducks/util.js [hunk 2]
```diff
@@ -1,7 +1,11 @@
 import app from 'ampersand-app';
 
+export const logUI = (level, msg) => {
+  app.renderLogger[level](msg);
```
Comment (swashcap) [inline packages/coinstac-ui/app/render/state/ducks/util.js:4]: Why expose  the logger as a two-argument function as opposed the default level-methods? Calls look cleaner:

```js
import { logUI as log } from '../state/ducks/util';
// ...
log('Imma message');
log.error('Oh noes!');
```

…versus:

```js
import { logUI as log from '../state/ducks/util';
// ...
log('info', 'Imma message');
log('error', 'Oh noes!');
```

This also means that one must look for two cases when attempting to track down log calls (`error`, for example: `log\.error` and `log\('error'`).

File: packages/coinstac-ui/app/main/utils/boot/configure-logger.js [hunk 3]
```diff
@@ -26,29 +36,39 @@ module.exports = function configureLogger() {
     });
 
     if (cliOpts.loglevel) {
```
Comment (swashcap) [inline packages/coinstac-ui/app/main/utils/boot/configure-logger.js:38]: Not sure we ever use CLI-based log levels. Wonder if we need it.

File: packages/coinstac-ui/app/render/state/ducks/util.js (hunk 2 above)
Comment (swashcap) [reply packages/coinstac-ui/app/render/state/ducks/util.js:4]: Also, `logUI` is a bit redundant: we’re in the `render` directory, and these calls happen from modules containing user interface components. How does simply `log` sound?

File: packages/coinstac-ui/app/render/utils/boot/configure-logger.js [hunk 4]
```diff
@@ -11,20 +12,29 @@ const mainLogger = electron.remote.require('app/main/utils/expose-app.js')().log
...
+function getLogger(consoleMethod, winstonMethod, logger) {
   return (...args) => {
     console[consoleMethod].apply(console, args); // eslint-disable-line no-console
-    mainLogger[winstonMethod].apply(null, ['ui', ...args]);
+    logger[winstonMethod].apply(null, ['ui', ...args]);
   };
 }
 
 module.exports = function configureLogger() {
-  app.logger = {
```
Comment (swashcap) [inline packages/coinstac-ui/app/render/utils/boot/configure-logger.js:22]: Can’t remember why we’re doing this. Seems pretty convoluted: a separate logger (wrapper?) for every method? This essentially decorates Winston methods. Wonder if there's another way around it.

File: packages/coinstac-ui/app/render/state/ducks/util.js (hunk 2 above)
Comment (swashcap) [reply packages/coinstac-ui/app/render/state/ducks/util.js:4]: _Also_ also, the render process’s logger still lives in the main process, and log calls are synchronous over RPC. Wonder if there's a better approach for this.

**Review by swashcap — CHANGES_REQUESTED** [review]

**Comment by rssk** [comment]
What's up w/ this @tkah ?

**Pull request closed without merging**